src/CareTogether.Core/Utilities/AppendBlobMultitenantEventLog.cs

```diff
@@ -0,0 +1,96 @@
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Newtonsoft.Json;
+using OneOf;
+using OneOf.Types;
+
+
+namespace CareTogether.Utilities
+{
+    public class AppendBlobMultitenantEventLog<T>
+    {
+        private LogType _logType;
+        private BlobServiceClient _blobServiceClient;
+        private ConcurrentDictionary<(Guid organizationId, Guid locationId), int> _tenantEventCounts;
+
+        public AppendBlobMultitenantEventLog(BlobServiceClient blobServiceClient, LogType logType)
+        {
+            this._blobServiceClient = blobServiceClient;
+            this._logType = logType;        
+        }
+
+        public async Task<OneOf<Success, Error>> AppendEventAsync(Guid organizationId, Guid locationId, T domainEvent, long expectedSequenceNumber)
+        {
+            var tenantContainer = _blobServiceClient.GetBlobContainerClient(organizationId.ToString());
+
+            var numberOfTenantEvents = _tenantEventCounts[(organizationId, locationId)];
+
+            var currentBlobNum = (50000 / numberOfTenantEvents)+1;
+
+            var tenantBlob = tenantContainer.GetAppendBlobClient($"{locationId}/{_logType.ConvertToString()}/{currentBlobNum.ToString("D5")}.ndjson");
+
+            if(!tenantBlob.Exists())
+            {
+                await tenantBlob.CreateAsync();
+                await tenantBlob.SetHttpHeadersAsync(new BlobHttpHeaders() { ContentType = "application/x-ndjson" });
+            }
+
+            var eventJson = JsonConvert.SerializeObject(domainEvent);
+
+            var cleanEventJson = eventJson.Replace("\n", "").Replace("\r", "");
+
+            var eventStream = new MemoryStream(Encoding.UTF8.GetBytes(cleanEventJson));
+
+            var appendReturn = (BlobAppendInfo) await tenantBlob.AppendBlockAsync(eventStream);
+            
+            if(appendReturn.BlobCommittedBlockCount == expectedSequenceNumber)
+            {
+                return new Success();
+            }
+            else
+            {
+                return new Error();
+            }
+        }
+
+        public async IAsyncEnumerable<(T DomainEvent, long SequenceNumber)> GetAllEventsAsync(Guid organizationId, Guid locationId)
+        {
+            var tenantContainer = _blobServiceClient.GetBlobContainerClient(organizationId.ToString());
+
+            var tenantBlobs = tenantContainer.GetBlobs(BlobTraits.None, BlobStates.None, $"{locationId}/{_logType.ConvertToString()}");
+
+            foreach (BlobItem blob in tenantBlobs)
+            {
+                var appendBlob = tenantContainer.GetAppendBlobClient(blob.Name);
+   
+                var eventStream = await appendBlob.OpenReadAsync();
+
+                string eventString = new StreamReader(eventStream).ReadToEnd();
+
+                using (StringReader reader = new StringReader(eventString))
+                {
+                    string line;
+                    long lineNumber = 0;
```

Comment:
There's a bug here: the `SequenceNumber` you're returning is not the same thing as a per-log-segment line number. Change this to `eventSequenceNumber` and initialize it outside of the `foreach` loop.